src/collection.js

```diff
@@ -149,7 +156,11 @@ export default class Collection {
    */
   createRecord(record, options={}) {
     const reqOptions = this._collOptions(options);
-    const request = requests.createRecord(this.name, record, reqOptions);
+    const { permissions } = reqOptions;
+    // XXX throw if bucket is undefined
+    const path = record.id ? endpoint("record", this.bucket.name, this.name, record.id) :
+                             endpoint("records", this.bucket.name, this.name);
```

Comment:
Same remark as previously, when `id` is `undefined` the `endpoint` function will see it as a non-passed arg.